plugins/org.eclipse.sirius.diagram.sequence/src/org/eclipse/sirius/diagram/sequence/business/internal/elements/SequenceDiagram.java

```diff
@@ -733,6 +737,44 @@ public ISequenceEvent apply(View from) {
         return allDelimitedEvents;
     }
 
+    /**
+     * Returns all combined fragments in the given diagram.
+     * 
+     * @return all combined fragments on the given diagram.
+     */
+    public Set<InteractionContainer> getAllInteractionContainers() {
+        List<InteractionContainer> allInteractionContainers = null;
+        LinkedHashSet<InteractionContainer> allOrderedInteractionContainers = null;
+        if (useCache) {
+            // Initialize from cache
+            if (allInteractionContainersCache != null) {
+                allInteractionContainers = new ArrayList<InteractionContainer>(allInteractionContainersCache);
+            }
+            allOrderedInteractionContainers = allOrderedInteractionContainersCache;
+        }
+        if (allOrderedInteractionContainers == null) {
+            if (allInteractionContainers == null) {
+                allInteractionContainers = new ArrayList<>();
+                for (Node node : Iterables.filter(Iterables.filter(getNotationDiagram().getChildren(), Node.class), InteractionContainer.notationPredicate())) {
```

Comment:
I guess you copied the code from the CombinedFragment but we probably should replace the guava utils by the java lambda

Reply:
I let this code for now. We should refactor the whole class to remove guava dependency